### PR TITLE
Enhancement:animation family loaded as standing (abc) uses "use file sequence"

### PR DESCRIPTION
**Pull request by FadyFS**
## Changelog Description
The changes are the following. We started by updating the the is_sequence(files) function allowing it to return True for a list of files which has only one file, since our animation in this provides just one alembic file. For the correct FPS number, we got the fps from the published ass/abc from the version data.

## Testing notes:
Load the animation with abc format with Maya

### CODE REVIEW

File: openpype/hosts/maya/plugins/load/load_arnold_standin.py [hunk 1]
```diff
@@ -83,6 +85,7 @@ def load(self, context, name, namespace, options):
             cmds.setAttr(standin_shape + ".dso", path, type="string")
             sequence = is_sequence(os.listdir(os.path.dirname(self.fname)))
             cmds.setAttr(standin_shape + ".useFrameExtension", sequence)
+            cmds.setAttr(standin_shape + ".abcFPS", 30)
```
Comment (BigRoy) [inline openpype/hosts/maya/plugins/load/load_arnold_standin.py:88]: Wait a second - this doesn't sound right?

Comment (FadyFS) [reply openpype/hosts/maya/plugins/load/load_arnold_standin.py:88]: I am not sure about this one, but it works for this specific animation. Let me know what you think

Comment (FadyFS) [reply openpype/hosts/maya/plugins/load/load_arnold_standin.py:88]: Thank you for your reply, I will look it up more thoroughly !

File: openpype/hosts/maya/plugins/load/load_arnold_standin.py [hunk 2]
```diff
@@ -84,6 +96,10 @@ def load(self, context, name, namespace, options):
             sequence = is_sequence(os.listdir(os.path.dirname(self.fname)))
             cmds.setAttr(standin_shape + ".useFrameExtension", sequence)
 
+            cmds.setAttr(standin_shape + ".dso", path, type="string")
+            matching_fps = get_fps(os.listdir(os.path.dirname(self.fname)))
+            cmds.setAttr(standin_shape + ".abcFPS", matching_fps)
```
Comment (moonyuet) [inline openpype/hosts/maya/plugins/load/load_arnold_standin.py:101]:
```suggestion
            fps = float(version["data"]["fps"])
            cmds.setAttr(standin_shape + ".abcFPS", fps)
```
you can get the fps from the published ass/abc from the version data.

File: openpype/hosts/maya/plugins/load/load_arnold_standin.py [hunk 3]
```diff
@@ -6,22 +6,34 @@
...
+    if len(files) == 1:
+        collections, remainder = clique.assemble(files, minimum_items=1)
+    else:
+        collections, remainder = clique.assemble(files)
     if collections:
         sequence = True
-
     return sequence
 
+def get_fps(standin_shape):
```
Comment (moonyuet) [inline openpype/hosts/maya/plugins/load/load_arnold_standin.py:31]: You dont need this function if you get the fps data from the published object through the version data.

File: openpype/hosts/maya/plugins/load/load_arnold_standin.py (hunk 2 above)
Comment (BigRoy) [reply openpype/hosts/maya/plugins/load/load_arnold_standin.py:101]: Note that technically a version **could** be lacking fps data, right? E.g. a publish from standalone publisher? (I suspect that doesn't know anything about its FPS?)

Comment (moonyuet) [reply openpype/hosts/maya/plugins/load/load_arnold_standin.py:101]: Something like this:(pesudo code only)
```
            has_published_fps = version["data"]["fps"]
            matching_fps = None
            if has_published_fps:
                matching_fps = float(version["data"]["fps"])
            else:
                matching_fps = get_fps(os.listdir(os.path.dirname(self.fname)))
            cmds.setAttr(standin_shape + ".abcFPS", matching_fps)
```

**Review by moonyuet — CHANGES_REQUESTED** [review]
see comments above

**Review by moonyuet — APPROVED** [review]
the code looks good....it is ready for the test

**Comment by ClementHector** [comment]
@FadyFS you can switch your PR to ready to review and set the finished comments to resolved 🥇

File: openpype/hosts/maya/plugins/load/load_arnold_standin.py [hunk 4]
```diff
@@ -84,6 +93,11 @@ def load(self, context, name, namespace, options):
             sequence = is_sequence(os.listdir(os.path.dirname(self.fname)))
             cmds.setAttr(standin_shape + ".useFrameExtension", sequence)
 
+            fps = float(version["data"].get("fps"))
+            if fps is None:
+                fps = get_current_fps()
+            cmds.setAttr(standin_shape + ".abcFPS", fps)
```
Comment (ClementHector) [inline openpype/hosts/maya/plugins/load/load_arnold_standin.py:99]: it is equivalent to:
```suggestion
            fps = float(version["data"].get("fps"), get_current_fps())
            cmds.setAttr(standin_shape + ".abcFPS", fps)
```

**Review by ClementHector — CHANGES_REQUESTED** [review]

**Review by moonyuet — COMMENTED** [review]
image
looks good, i also did some simple test with the loader in maya(without fps data) (it might need some more complicated tests to see if it breaks anything)
it loaded with the same data regarding to the avalon.session.

File: openpype/hosts/maya/plugins/load/load_arnold_standin.py [hunk 5]
```diff
@@ -6,23 +6,29 @@
...
-    collections, remainder = clique.assemble(files)
+    collections, remainder = clique.assemble(files, minimum_items=1)
     if collections:
         sequence = True
-
     return sequence
 
 
+def get_current_session_fps():
+    session_fps = float(legacy_io.Session.get('AVALON_FPS', 25))
```
Comment (BigRoy) [inline openpype/hosts/maya/plugins/load/load_arnold_standin.py:29]: @iLLiCiTiT is this `legacy_io` usage ok, or is there already an alternative to this?

**Review by BigRoy — APPROVED** [review]
Didn't do a test run, but the code changes seem fine to me. Nice work.

**Review by LiborBatek — APPROVED** [review]
Did test and loaded animation family as Ass(abc) and all good. 
It has correctly set "Use File Sequence" and updates fine when playback.

image

**Pull request merged**